src/renderer/app.tsx

```diff
@@ -1,32 +1,123 @@
+  };
+
+  return (
+    <Accordion activeKey={activeKeyState} className={theme}>
+      {tweets.map((tweet) => (
+        <Card key={tweet.id}>
+          <Card.Header>
+            <Accordion.Toggle
+              as={Button}
+              variant="link"
+              eventKey={tweet.id}
+              onClick={() => handleActiveKeyChange(tweet.id)}>
+              {tweet.text.split(' ').slice(0, 15).join(' ')}
+            </Accordion.Toggle>
+          </Card.Header>
+          <Accordion.Collapse eventKey={tweet.id}>
+            <div>
+              <Card.Body>{tweet.text.split(' ').slice(15).join(' ')}</Card.Body>
+              {tweet.videoPath ? <video controls playsInline src={tweet.videoPath} /> : <div></div>}
+            </div>
+          </Accordion.Collapse>
+        </Card>
+      ))}
+    </Accordion>
+  );
+};
+
 const App: FC = () => {
-  const [tweets, setTweets] = useState<Tweet[]>();
-
-  useEffect((): void => {
-    (async () => {
-      let fetchedTweets: Tweet[];
-      do {
-        fetchedTweets = await fetch('https://r-t-generator.herokuapp.com/').then(
-          (response) => response.json() as Promise<Tweet[]>,
-        );
-      } while (!fetchedTweets.length);
-      setTweets(fetchedTweets);
-      // eslint-disable-next-line no-console
-    })().catch(console.error);
-  }, []);
+  const [tweetsList, setTweets] = useState<Tweet[]>();
+  const [radioValue, setRadioValue] = useState('light');
+
+  const updateTweets = (tweets: Tweet[]) => {
+    fetch('https://r-t-generator.herokuapp.com/')
+      .then((response) => response.json())
+      .then((data) => {
+        let fetchedTweets: Tweet[] = data;
+        let i = 0;
+        for (i; i < fetchedTweets.length; i++) {
+          fetchedTweets[i].id = (i + (tweets ? tweets : []).length).toString(10);
+          fetchedTweets[i].videoPath = fetchedTweets[i].videoPath
+            ? 'https://r-t-generator.herokuapp.com/' + fetchedTweets[i].videoPath
+            : undefined;
+        }
+        fetchedTweets = fetchedTweets.concat(tweets ? tweets : []);
+        // console.log(fetchedTweets);
+        setTweets(fetchedTweets);
+      });
+  };
 
+  useEffect(() => {
+    const interval = setInterval((tweets: Tweet[] = tweetsList ? tweetsList : []) => {
+      updateTweets(tweets ? tweets : []);
+    }, 15000);
+    return () => clearInterval(interval);
+  }, [tweetsList]);
   return (
     <Fragment>
-      <div>Hello world!</div>
-      {tweets && <pre>{JSON.stringify(tweets, null, 4)}</pre>}
+      <ThemeSelector radioValue={radioValue} setRadioValue={setRadioValue} />
+      {tweetsExist(tweetsList, radioValue)}
     </Fragment>
   );
 };
 
+const tweetsExist = (tweetsList: Tweet[] | undefined, theme: string) => {
+  if (tweetsList && tweetsList.length > 0) {
+    return <TweetContainer tweets={tweetsList.length > 25 ? tweetsList.slice(25) : tweetsList} theme={theme} />;
+  } else {
+    return <Fragment> No Tweets From Server </Fragment>;
```

Comment:
I suppose just returning a string should also work? If so, let's not add unnecessary complexity